views/jira-connected-repos.hbs

```diff
@@ -0,0 +1,108 @@
+</head>
+
+<body class="aui-page-hybrid">
+<div class="jiraConnectedRepos">
+    <div class="jiraConnectedRepos__header">
+        <div>
+            <button class="navigation__backButton__container navigation__link go-back">
+                <img src="/public/assets/arrow-left.svg" alt="Go back" />
+            </button>
+        </div>
+        <div class="jiraConnectedRepos__header__title">
+            <h1>
+                GitHub connected repositories
+            </h1>
+        </div>
+    </div>
+
+    <section class="jiraConnectedRepos_content">
+
+        <div class="jiraConnectedRepos__tableContainer">
+            <table class="jiraConnectedRepos__table aui">
+                <thead class="jiraConnectedRepos__table__head">
+                <tr class="jiraConnectedRepos__table__head_row">
+                    <th class="jiraConnectedRepos__table__head__title">Repository</th>
+                    <th class="jiraConnectedRepos__table__head__title">Backfill</th>
+                </tr>
+                </thead>
+
+                <tbody class="jiraConnectedRepos__table__body">
+                {{#each repos}}
+                <tr class="jiraConnectedRepos__table__row">
+
+                    {{! Repo name }}
+                    <td class="jiraConnectedRepos__table__cell">
+                        {{name}}
+                    </td>
+
+                    {{! Backfill status }}
+                    <td class="jiraConnectedRepos__table__cell">
+                        <div class="jiraConnectedRepos__table__cell__syncStatus">
+                            <span class="jiraConnectedRepos__table__{{toLowerCase
+                                                    (replaceSpaceWithHyphen syncStatus)
+                                                  }}
+                                                  jiraConnectedRepos__table__syncStatus">
+                            {{syncStatus}}
+                            </span>
+                        </div>
+                    </td>
+
+                </tr>
+                {{/each}}
+                </tbody>
+            </table>
+        </div>
+        <div class="jiraConnectedRepos_pagination" data-subscription-id="{{subscriptionId}}">
+            {{#if hasPrevPage}}
+            <a href="#" class="prev-page page-selector" data-page-num="{{prevPageNum}}">Prev</a>
+            {{/if}}
+
+            {{#each pages}}
+                {{#if isCurrentPage}}
+                <span class="current-page">{{pageNum}}</span>
+                {{else}}
+                <a href="#" class="page-num-link page-selector" data-page-num="{{pageNum}}">{{pageNum}}</a>
+                {{/if}}
+            {{/each}}
+
+            {{#if hasNextPage}}
+            <a href="#" class="next-page page-selector" data-page-num="{{nextPageNum}}">Next</a>
+            {{/if}}
+        </div>
+    </section>
+
+<!--    <span class="pageone" data-subscription-id="{{subscriptionId}}">PAGE 1</span>-->
```

Comment:
Can we delete this?

Reply:
deleted

Reply:
> Should we feature this? I can see we limit it to 10 per page but it doesn't look like we're limiting the number of repos we're requesting at a time. Worried things could blow up/really slow down.
> 
> Also, have we thought about what the UI should look like if we have more than 4 pages? What if I have 100 pages? Are we planning to render all numbers from 1 to 100? And I can see we hide prev/next if there isn't a previous next - does this mean the numbers move around? i.e. I'm on page 1 and 'Prev' is hidden but then I click page 2 and suddenly 'Prev' appears and all numbers shift to the right? A recording of this would be great but good UX for implementing pagination is to disable prev/next as opposed to hiding/displaying them.

The UI can handle as many pages as you can throw at it and it will show a nuch before and after e.g. 2000 pages on page 600 will show prev 598 599 600 601 602 next 

We only send one page of data to the template at a time so the performance should be ok since we retrieve from our db, however, we are changing it to a seperate API fetch so it can be fine tuned a bit better.

prev and next sometime showing, numbers bouncing... yup, agree its pretty lame, ill style it so its a fixed position for the numbers and Sarah has shown us a prettier way to do it.

If you have 329083479 Repos its simply unusable, so will be adding a filter to show/hide based of status and also a search.